fetching/strat.py

```diff
@@ -29,8 +35,61 @@ def _test_file_filter(self, path: pathlib.Path) -> bool:
 
 
 class PyTestStrategy(ApplicationStrategy):
+    FUNCTION_PATTERN = re.compile(r"[\w\s]*test_[\w\s]*\([\w\s]*\)[\w\s]*:[\w\s]*")
+    IMPORTS = "from tracing import register, entrypoint\n"
+    REGISTER = "@register()\n"
+    ENTRYPOINT = "\n@entrypoint()\ndef main():\n  ...\n"
+    APPENDED_FILEPATH = "_decorators_appended.py"
+
+    def __init__(self, recurse_into_subdirs: bool = True):
+        super().__init__(recurse_into_subdirs)
+
+        self.decorator_appended_file_paths: list[pathlib.Path] = []
+
     def _apply(self, path: pathlib.Path) -> None:
-        pass
+        with path.open("r") as file:
+            lines = file.readlines()
+
+        skip_line = False
+        contains_pytest_test_function = False
+        for i, line in enumerate(lines):
+            if skip_line:
+                skip_line = False
+                continue
+            if PyTestStrategy.FUNCTION_PATTERN.fullmatch(line):
+                lines.insert(i, PyTestStrategy.REGISTER)
+                skip_line = True
+                contains_pytest_test_function = True
+
+        if contains_pytest_test_function:
+            lines.insert(0, PyTestStrategy.IMPORTS)
+            lines.append(PyTestStrategy.ENTRYPOINT)
+
+        file_path_with_appended_decorators = pathlib.Path(
+            str(path).replace(".py", PyTestStrategy.APPENDED_FILEPATH)
+        )
+        with file_path_with_appended_decorators.open("w") as file:
+            file.writelines(lines)
+
+        self.decorator_appended_file_paths.append(file_path_with_appended_decorators)
 
     def _test_file_filter(self, path: pathlib.Path) -> bool:
```

Comment:
I recommend renaming it to _is_file_valid_for_strategy or something similar. I would not expect this to return a bool with the current name.

Reply:
Good point

Reply:
Method name is now `_is_test_file`